$NetBSD: patch-src_xercesc_util_Xerces__autoconf__config.hpp.cmake.in,v 1.2 2025/01/31 15:13:26 wiz Exp $

These symbols may be undefined, so check for them with #ifdef.
https://issues.apache.org/jira/browse/XERCESC-2260

--- src/xercesc/util/Xerces_autoconf_config.hpp.cmake.in.orig	2025-01-31 12:07:26.048504288 +0000
+++ src/xercesc/util/Xerces_autoconf_config.hpp.cmake.in
@@ -95,16 +95,16 @@
 #if defined(__cplusplus) && defined(XERCES_HAVE_CSTDINT)
 #  include <cstdint>
 #endif
-#if XERCES_HAVE_STDINT_H
+#ifdef XERCES_HAVE_STDINT_H
 #  include <stdint.h>
 #endif
-#if XERCES_HAVE_INTTYPES_H
+#ifdef XERCES_HAVE_INTTYPES_H
 #  include <inttypes.h>
 #endif
-#if XERCES_HAVE_SYS_TYPES_H
+#ifdef XERCES_HAVE_SYS_TYPES_H
 #  include <sys/types.h>
 #endif
-#if XERCES_INCLUDE_WCHAR_H
+#ifdef XERCES_INCLUDE_WCHAR_H
 #  include <wchar.h>
 #endif
 
